src/sage/rings/function_field/drinfeld_modules/drinfeld_module.py

```diff
@@ -1089,19 +1400,137 @@ def j_invariant(self):
+            True
+
+        The list of all basic `j`-invariant parameters can be retrieved
+        using the method :meth:`basic_j_invariant_parameters`::
+
+            sage: A = GF(3)['T']
+            sage: K.<T> = Frac(A)
+            sage: phi = DrinfeldModule(A, [T, T^2 + T + 1, 0, T^4 + 1, T - 1])
+            sage: param = phi.basic_j_invariant_parameters(nonzero=True)
+            sage: phi.j_invariant(param[0])
+            T^13 + 2*T^12 + T + 2
+            sage: phi.j_invariant(param[1])
+            T^35 + 2*T^31 + T^27 + 2*T^8 + T^4 + 2
+
+        TESTS::
+
+            sage: A = GF(5)['T']
+            sage: K.<T> = Frac(A)
+            sage: phi = DrinfeldModule(A, [T, T^2, 1, T + 1, T^3])
+            sage: phi.j_invariant()
             Traceback (most recent call last):
             ...
-            NotImplementedError: rank must be 2
+            TypeError: parameter must be different from NoneType if the rank is greater than 2
+            sage: phi.j_invariant(-1)
+            Traceback (most recent call last):
+            ...
+            ValueError: integer parameter must be >= 1 and < the rank (=4)
+            sage: phi.j_invariant('x')
+            Traceback (most recent call last):
+            ...
+            TypeError: parameter must be a tuple or a list of length 2 or an integer
+            sage: phi.j_invariant((1, 2, 3))
+            Traceback (most recent call last):
+            ...
+            ValueError: list or tuple parameter must be of length 2
+            sage: phi.j_invariant(('x', (1, 2, 3)))
+            Traceback (most recent call last):
+            ...
+            TypeError: list or tuple parameter must contain tuples or lists
+            sage: phi.j_invariant(((1, 2), 'x'))
+            Traceback (most recent call last):
+            ...
+            TypeError: list or tuple parameter must contain tuples or lists
+            sage: phi.j_invariant(((1, 2, 3, 4, 5), (2, 1)))
+            Traceback (most recent call last):
+            ...
+            ValueError: components of tuple or list parameter have incorrect length
+            sage: phi.j_invariant(((1, 'x'), (2, 3, 8)))
+            Traceback (most recent call last):
+            ...
+            TypeError: components of tuple or list parameter must contain only integers
+            sage: phi.j_invariant(((1, 2), (2, 3, 'x')))
+            Traceback (most recent call last):
+            ...
+            TypeError: components of tuple or list parameter must contain only integers
+            sage: phi.j_invariant(((1, 2), (4, 3, 7)))
+            Traceback (most recent call last):
+            ...
+            ValueError: parameter does not satisfy the weight-0 condition
+            sage: phi.j_invariant(((1, 2), (4, 3, 7)), check=False)
+            1/T^13
         """
-        self._check_rank_two()
-        g = self.coefficient(1)
-        delta = self.coefficient(2)
+        r = self._gen.degree()
```

Comment:
Again, I would recommend using

```suggestion
        r = self.rank()
```